slaves/linux-cross/Dockerfile

```diff
@@ -38,6 +36,30 @@ RUN pip install buildbot-slave
 RUN groupadd -r rustbuild && useradd -r -g rustbuild rustbuild
 RUN mkdir /buildslave && chown rustbuild:rustbuild /buildslave
 
+# Install arm cross compiler
+WORKDIR /build
+COPY linux-cross/build_arm_toolchain_root.sh /build/
+RUN /bin/bash build_arm_toolchain_root.sh && chown rustbuild:rustbuild /build
+RUN mkdir /x-tools && chown rustbuild:rustbuild /x-tools
+COPY linux-cross/build_arm_toolchain.sh /build/
+COPY linux-cross/arm-linux-gnueabi.config /build/
+COPY linux-cross/arm-linux-gnueabihf.config /build/
+USER rustbuild
+RUN /bin/bash build_arm_toolchain.sh arm-linux-gnueabi
+RUN /bin/bash build_arm_toolchain.sh arm-linux-gnueabihf
+USER root
+RUN rm -rf /build
```

Comment:
I've found that not having a bunch of intermediate artifacts can hugely reduce the size of docker containers, could this be done as part of each step? Some examples I've done in the past are:

```
RUN /bin/bash build_arm_toolchain.sh arm-linux-gnueabi && rm -rf /build
```

(or something like that)